x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -649,14 +649,19 @@ static DeprecationIssue checkSingleDataNodeWatermarkSetting(
         if (DiskThresholdDecider.ENABLE_FOR_SINGLE_DATA_NODE.get(settings) == false
             && DiskThresholdDecider.ENABLE_FOR_SINGLE_DATA_NODE.exists(settings)) {
             String key = DiskThresholdDecider.ENABLE_FOR_SINGLE_DATA_NODE.getKey();
+            String disableDiskDecider = DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_THRESHOLD_ENABLED_SETTING.getKey();
             return new DeprecationIssue(
                 DeprecationIssue.Level.CRITICAL,
                 String.format(Locale.ROOT, "Setting [%s=false] is deprecated", key),
                 "https://ela.st/es-deprecation-7-disk-watermark-enable-for-single-node-setting",
                 String.format(
                     Locale.ROOT,
-                    "Remove the [%s] setting. Disk watermarks are always enabled for single node clusters in 8.0.",
-                    key
+                    "Disk watermarks do not treat single-node clusters differently in versions 8.0 and later, and [%s] may not be set to "
+                        + "[false] in these versions. Set [%s] to [true] to adopt the future behavior before upgrading. "
+                        + "If desired you may also set [%s] to [false] to completely disable disk-based allocation.",
```

Comment:
I wonder if we should avoid this part of the message, since it also disables flood stage monitoring? We made this setting an operator-only setting too, i.e., may not be settable by users in all environments.

Same comment for all 4 checks.

Reply:
Sure, that seems even better to me.